src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx

```diff
@@ -101,9 +100,58 @@ const RuleEditor = createReactClass({
     Promise.all(promises).then(([environments, rule]) => {
       this.setState({environments, rule});
     });
-  },
+  }
+
+  fetchStatus() {
+    this.pollHandler();
+  }
+
+  pollHandler = async () => {
+    const {api, organization, project} = this.props;
+    const {uuid} = this.state;
+    const origRule = {...this.state.rule};
+
+    const response = await api
+      .requestPromise(`/projects/${organization.slug}/${project.slug}/rule-task/${uuid}/`)
+      .catch(() => {
+        addErrorMessage(t('An error occurred'));
+        this.setState({loading: false});
+      });
+
+    const {status, rule, error} = response;
+
+    if (status === 'pending') {
+      setTimeout(this.pollHandler, 1000);
+      return;
+    }
+
+    if (status === 'failed') {
+      this.setState({
+        error: {actions: [error]} || {__all__: 'Unknown error'},
+        loading: false,
+      });
+      addErrorMessage(t('An error occurred'));
+    }
+    if (rule && status === 'success') {
+      const isNew = !origRule.id;
+      this.handleRuleSuccess(isNew, rule);
+    }
+  };
+
+  startPolling() {
+    setTimeout(this.pollHandler, 1000);
```

Comment:
@MeredithAnya I am still a bit concerned this polling could go on forever. I think we need to give up at some point if we don't get a response.